elki/src/main/java/elki/math/statistics/dependence/MCDEDependenceMeasure.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * This file is part of ELKI:
+ * Environment for Developing KDD-Applications Supported by Index-Structures
+ *
+ * Copyright (C) 2019
+ * ELKI Development Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package elki.math.statistics.dependence;
+
+import elki.utilities.datastructures.arraylike.NumberArrayAdapter;
+import elki.utilities.documentation.Reference;
+import elki.utilities.exceptions.AbortException;
+import elki.utilities.optionhandling.OptionID;
+import elki.utilities.optionhandling.Parameterizer;
+import elki.utilities.optionhandling.constraints.CommonConstraints;
+import elki.utilities.optionhandling.parameterization.Parameterization;
+import elki.utilities.optionhandling.parameters.DoubleParameter;
+import elki.utilities.optionhandling.parameters.IntParameter;
+import elki.utilities.optionhandling.parameters.ObjectParameter;
+import elki.utilities.optionhandling.parameters.RandomParameter;
+import elki.utilities.random.RandomFactory;
+import elki.math.statistics.tests.mcde.*;
+
+import java.util.Arrays;
+import java.util.Random;
+
+/**
+ * Implementation of bivariate Monte Carlo Density Estimation as described in
+ * Edouard Fouché & Klemens Böhm<br>
```

Comment:
The first sentence of a JavaDoc should be a short regular sentence (because it will be used as a short description of the class). We usually have the references at the end of the JavaDoc because of this.